Lib/logging/__init__.py

```diff
@@ -1570,6 +1570,13 @@ def __repr__(self):
         level = getLevelName(self.getEffectiveLevel())
         return '<%s %s (%s)>' % (self.__class__.__name__, self.name, level)
 
+    def __reduce__(self):
+        # In general, only the root logger will not be accessible via its name.
+        # However, the root logger's class has its own __reduce__ method.
+        if getLogger(self.name) is not self:
+            raise pickle.PicklingError('logger cannot be pickled')
```

Comment:
The `pickle` module is used only here. It isn't used if pickling is not involved. May be make an exception from PEP 8 and import it locally?